schema/migration-2-0029-20231017.sql

```diff
@@ -0,0 +1,40 @@
+-- Persistent generated migration.
+
+CREATE FUNCTION migrate() RETURNS void AS $$
+DECLARE
+  next_version int ;
+BEGIN
+  SELECT stage_two + 1 INTO next_version FROM schema_version ;
+  IF next_version = 29 THEN
+    -- if the table "pool_offline_data" all ready exists just rename everything
+    IF EXISTS (SELECT 1 FROM information_schema.tables WHERE table_name = 'pool_offline_data') THEN
+      EXECUTE 'ALTER TABLE "pool_offline_data" RENAME TO "off_chain_pool_data"' ;
+      EXECUTE 'ALTER TABLE "off_chain_pool_data" RENAME CONSTRAINT "unique_pool_offline_data" TO "unique_off_chain_pool_data"' ;
+      EXECUTE 'ALTER TABLE "off_chain_pool_data" RENAME CONSTRAINT "pool_offline_data_pkey" TO "off_chain_pool_data_pkey"' ;
+    -- otherwise we create the newly named table and it's constraint "off_chain_pool_data"
+    ELSE
```

Comment:
I think only this `else` case will ever hit, both for migrations from existing dbs and from empty. So postgres will attempt to recreate an existing table.